doc/dynamo0p3.rst

```diff
@@ -891,22 +894,34 @@ as the number of dofs for each of the dofmaps. The full set of rules is:
 	  7) Include banded-matrix parameter ``gamma_p``. This is an integer
 	     with intent ``in`` and is named as ``"gamma_p_"<operator_name>``.
 
-    6) Include the required dofmaps and their dimensions:
-
-       1) Include ``ndf_to``, the number of degrees of freedom per cell for
-	  the to-space of the CMA operator. This is an integer with intent
-	  ``in``.
-       2) If the from-space of the operator is *not* the same as the 
-	  to-space then include ``ndf_from``, the number of degrees of
-	  freedom per cell for the from-space of the CMA operator. This is
-	  an integer with intent ``in``.
-       3) Include ``column_banded_dofmap_to``, the list of offsets for the
-	  to-space. This is an integer array of rank 2. The first dimension
-	  is ``ndf_to`` and the second is ``nlayers``.
-       4) If the from-space of the operator is *not* the same as the 
-	  to-space then include ``column_banded_dofmap_from``, the list of
-	  offsets for the from-space. This is an integer array of rank 2. The
-	  first dimension is ``ndf_from`` and the second is ``nlayers``.
+       3) If it is a field or scalar argument then include arguments following
+          the same rules as for general-purpose kernels. 
+	  
+    6) For each unique function space in the order they appear in the
+       metadata arguments (the ``to`` function space of an operator is
+       considered to be before the ``from`` function space of the same
+       operator as it appears first in lexicographic order):
+
+       1) Include the number of degrees of freedom per cell for
+	  the space. This is an integer with intent ``in``. The name of this
+	  argument is ``"ndf_"<arg_function_space>``.
+
+       2) If there is a field on this space then:
+
+	  1) Include the unique number of degrees of freedom for the
+ 	     function space. This is an integer and has intent ``in``.
+             The name of this argument is ``"undf_"<field_function_space>``.
+
+	  2) Include the dofmap for this space. This is an integer array
+	     with intent ``in``. It has one dimension sized by the local
+	     degrees of freedom for the function space.
+
+	3) If the CMA operator has this space as its to/from space then
+	   include the column-banded dofmap, the list of offsets for the
+	   to/from-space. This is an integer array of rank 2. The first
+	   dimension is ``"ndf_"<arg_function_space>```` and the second
+           is ``nlayers``.
+
 
 Application/Inverse-Application
```

Comment:
I thought that a dofmap is only ever for a single cell since we have contiguous data-layout in the vertical. i.e. kernels do `map(df)+k` to get the appropriate array index for dof `df` on level `k`. We don't pass in `nlevels` because we the CMA operator acts on whole columns. I've added a line to clarify.